packages/dds/tree/src/test/forestTestSuite.ts

```diff
@@ -85,1032 +84,999 @@ const buildId = { minor: 42 };
+	});
 
-			const parentAnchor = forest.anchors.track(parentPath);
-			const childAnchor1 = forest.anchors.track(childPath1);
-			const childAnchor2 = forest.anchors.track(childPath2);
+	it("anchors creation and use", () => {
+		const forest = factory(new TreeStoredSchemaRepository(optionalArraySchema));
+		initializeForest(
+			forest,
+			fieldJsonCursor([[1, 2]]),
+			testRevisionTagCodec,
+			testIdCompressor,
+		);
+
+		const cursor = forest.allocateCursor();
+		moveToDetachedField(forest, cursor);
+		assert(cursor.firstNode());
+		const parentAnchor = cursor.buildAnchor();
+		cursor.enterField(EmptyKey);
+		cursor.enterNode(0);
+		assert.equal(cursor.value, 1);
+		const childAnchor1 = cursor.buildAnchor();
+		assert(cursor.nextNode());
+		const childAnchor2 = cursor.buildAnchor();
+		cursor.exitNode();
+		cursor.exitField();
+		const parentAnchor2 = cursor.buildAnchor();
+
+		const parentPath = clonePath(forest.anchors.locate(parentAnchor));
+		const childPath1 = clonePath(forest.anchors.locate(childAnchor1));
+		const childPath2 = clonePath(forest.anchors.locate(childAnchor2));
+		const parentPath2 = clonePath(forest.anchors.locate(parentAnchor2));
+
+		const expectedParent: UpPath = {
+			parent: undefined,
+			parentField: rootFieldKey,
+			parentIndex: 0,
+		};
 
-			const cursor = forest.allocateCursor();
-			assert.equal(forest.tryMoveCursorToNode(parentAnchor, cursor), TreeNavigationResult.Ok);
-			assert.equal(cursor.value, undefined);
-			assert.equal(forest.tryMoveCursorToNode(childAnchor1, cursor), TreeNavigationResult.Ok);
-			assert.equal(cursor.value, 1);
-			assert.equal(forest.tryMoveCursorToNode(childAnchor2, cursor), TreeNavigationResult.Ok);
-			assert.equal(cursor.value, 2);
-		});
+		assert.deepStrictEqual(parentPath, expectedParent);
+		assert.deepStrictEqual(parentPath2, expectedParent);
+
+		const expectedChild1: UpPath = {
+			parent: expectedParent,
+			parentField: EmptyKey,
+			parentIndex: 0,
+		};
+
+		const expectedChild2: UpPath = {
+			parent: expectedParent,
+			parentField: EmptyKey,
+			parentIndex: 1,
+		};
+
+		assert.deepStrictEqual(childPath1, expectedChild1);
+		assert.deepStrictEqual(childPath2, expectedChild2);
+
+		assert.equal(forest.tryMoveCursorToNode(parentAnchor, cursor), TreeNavigationResult.Ok);
+		assert.equal(cursor.value, undefined);
+		assert.equal(forest.tryMoveCursorToNode(childAnchor1, cursor), TreeNavigationResult.Ok);
+		assert.equal(cursor.value, 1);
+		assert.equal(forest.tryMoveCursorToNode(childAnchor2, cursor), TreeNavigationResult.Ok);
+		assert.equal(cursor.value, 2);
+
+		// Cleanup is not required for this test (since anchor set will go out of scope here),
+		// But make sure it works:
+		forest.anchors.forget(parentAnchor);
+		forest.anchors.forget(childAnchor1);
+		forest.anchors.forget(childAnchor2);
+		forest.anchors.forget(parentAnchor2);
+		assert(forest.anchors.isEmpty());
+	});
+
+	it("using an anchor that went away returns NotFound", () => {
```

Comment:
Nit: "went away" is somewhat vague terminology. Is there something more specific we should say?

Reply:
improved